x-pack/plugins/maps/public/components/layer_panel/view.js

```diff
@@ -35,11 +35,16 @@ export class LayerPanel  extends React.Component {
         displayName: '',
         immutableSourceProps: [],
         hasLoadedSourcePropsForLayer: false,
+        styleUpdated: false,
+        style: nextProps.styleProperties,
         prevId: nextId,
       };
     }
-
-    return null;
+    return {
```

Comment:
remove this `styleUpdated` property from state. We only need the `stateChanged` property.